### DISCUSSION TITLE
[@types/react-dom] Wrong type for hydrate function.

### DISCUSSION THREAD

**Discussion by matej-marcisovsky** [DefinitelyTyped/DefinitelyTyped#61610] [Issues with a @types package]
Hi,

in my app, I am using `hydrate` from `react-dom`. In my code I am passing to `hydrate` a React element in variable called `reactElementView`. I get follwing error and can not figure out what I am doing wrong. If I use the same variable with `render` function from `react-dom`, everything is OK.

```
const reactElementView: React.ReactElement<{}, string | React.JSXElementConstructor<any>>
No overload matches this call.
  The last overload gave the following error.
    Argument of type 'ReactElement<{}, string | JSXElementConstructor<any>>' is not assignable to parameter of type 'ReactElement<any, string | JSXElementConstructor<any>>[]'.
      Type 'ReactElement<{}, string | JSXElementConstructor<any>>' is missing the following properties from type 'ReactElement<any, string | JSXElementConstructor<any>>[]': length, pop, push, concat, and 29 more.
```

I am using react 17.0.2, node 18.3.0, @types/react-dom 18.0.6

[a comment by an automated account is not shown]

**Comment by eps1lon** [comment]
You likely have multiple versions of `@types/react` installed. Having multiple versions installed is not supported and will lead to above errors. `npm ls @types/react` (when using NPM) or `yarn why @types/react` (when using Yarn) should only list a single installed version. To fix, run `npx yarn-deduplicate --packages @types/react` (when using Yarn). When using NPM the only choice is `npm dedupe` which de-duplicates all which might not be desired.

If this didn't help, please provide a minimal repro (e.g. a [TypeScript playground](typescriptlang.org/play))

**Comment by matej-marcisovsky** [comment]
Looks like I have only one version:

```
ima@
└─┬ @ima/react-page-renderer@18.0.0-rc.2 -> ./packages/react-page-renderer
  ├─┬ @types/react-dom@18.0.6
  │ └── @types/react@18.0.17 deduped
  └── @types/react@18.0.17
```

I will prepare a minimal repro to test it out.
